### PR TITLE
Added locale to redirect_to helper. Made use of it in the calls in re…

### PR DESCRIPTION
**Pull request by JVickery-TBS**
…quest_reset.

Fixes #

Password reset form redirect does not maintain current locale. It just redirects to "/". Added locale to helper method and made use of it in request_reset redirects. This is just a small hot fix which will most likely e removed soon from the upgrade to ckan2.9

### Features:

- [ ] includes tests covering changes
- [ ] includes updated documentation
- [ ] includes user-visible changes
- [ ] includes API changes
- [ ] includes bugfix for possible backport
- [X] None of the above

Please [X] all the boxes above that apply

### CODE REVIEW

File: ckan/controllers/user.py [hunk 1]
```diff
@@ -512,14 +512,13 @@ def request_reset(self):
                         _(u'Error sending the email. Try again later '
                           'or contact an administrator for help')
                     )
-                    log.exception(e)
```
Comment (wardi) [inline ckan/controllers/user.py:515]: better keep the log here

File: ckan/lib/helpers.py [hunk 2]
```diff
@@ -189,8 +189,12 @@ def redirect_to(*args, **kw):
     if skip_url_parsing is False:
         _url = url_for(*uargs, **kw)
 
+    locale = kw.pop('locale', '')
+    if len(locale) > 0:
```
Comment (wardi) [inline ckan/lib/helpers.py:193]:
```suggestion
    if locale:
```

**Comment by wardi** [comment]
looks like this is also an issue upstream (there's a `redirect_to('/user/reset')` in views/user.py and the helper code looks the same) also as a really minor point this change isn't handling root_path correctly.

Would you submit a similar fix to core instead so it can be discussed and assigned?

**Comment by JVickery-TBS** [comment]
@wardi okay this code should be good now. I added your suggestions (also oops I removed that log statement).

I also added the locale to that other `redirect_to('/user/reset')`

I will work on a PR for ckan/ckan master

**Comment by wardi** [comment]
looks good. Could the same be submitted upstream?

**Pull request merged**